src/ucs/sys/sys.c

```diff
@@ -442,6 +443,38 @@ size_t ucs_get_phys_mem_size()
     return phys_pages * ucs_get_page_size();
 }
 
+#define UCS_SHMMAX "/proc/sys/kernel/shmmax"
+size_t ucs_get_shmmax()
+{
+    char buf[256];
+    FILE *f;
+    char *out;
+    size_t size = 0;
+    int rc;
+
+    f = fopen(UCS_SHMMAX, "r");
```

Comment:
BTW, there are other places in the code where we have to move to ucs_read_file(). The full list of offenders:
ucs_get_huge_page_size, ucs_get_cpuinfo_clock_freq,ucs_debug_print_source_file, ucs_x86_tsc_freq_from_cpu_model